mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1100,18 +1164,14 @@ pub async fn validate_taker_funding_spend_preimage<T: UtxoCommonOps + SwapOps>(
         )));
     }
 
-    let expected_fee = coin
-        .get_htlc_spend_fee(DEFAULT_SWAP_TX_SPEND_SIZE, &FeeApproxStage::WithoutApprox)
-        .await?;
-
     let actual_fee = funding_amount - payment_amount;
+    let instructed_fee =
+        sat_from_big_decimal(&gen_args.taker_payment_fee, coin.as_ref().decimals).mm_err(|e| e.into())?;
 
-    let fee_div = expected_fee as f64 / actual_fee as f64;
-
-    if !(0.9..=1.1).contains(&fee_div) {
+    if actual_fee < instructed_fee {
         return MmError::err(ValidateTakerFundingSpendPreimageError::UnexpectedPreimageFee(format!(
-            "Too large difference between expected {} and actual {} fees",
-            expected_fee, actual_fee
+            "A fee of {} was used, less than the agreed upon fee of {}",
+            actual_fee, instructed_fee
```

Comment:
Ooh, I see why you used the max of `calculated_fee` and `taker_instructed_fee` above. I don't think this validation is actually needed at all, this is done from the taker side, and once the swap starts any additional fee or reduced fee will either reduce the maker amount or benefit him so the maker can choose any fee on his side when generating the preimage and the taker doesn't have to validate it. Also for the below, maker negotiates a fee at the start / negotiation phase and taker either approves it or not, so there is no room for abuse.
https://github.com/KomodoPlatform/komodo-defi-framework/blob/b6ee6984c2f5fcceb6ded7ad185ff5cd0221dce7/mm2src/coins/utxo/utxo_common.rs#L1075

Reply:
> once the swap starts any additional fee or reduced fee will either reduce the maker amount or benefit him so the maker can choose any fee on his side when generating the preimage and the taker doesn't have to validate it

I guess we have to lookout for the refund case though, since if maker chooses a very high fee and doesn't complete the swap, taker will lose funds on refund.

Reply:
> Also for the below, maker negotiates a fee at the start / negotiation phase and taker either approves it or not, so there is no room for abuse.

What about removing the fee pumping logic (so we won't calculate `calculated_fee` and won't use the `max`) and use the `taker_instructed_fee` right away. Re-thinking about it, the `max`-ing might not make that much of a difference (*most of the time*), and this would also stop the attack vector where the maker sets a high fee just to hit the taker on his refund.

Reply:
Using the `taker_instructed_fee` is the right idea IMHO but the problem is fees spikes as always. There are 2 solutions:

1. Either just use `taker_instructed_fee`, taker will want to optimize for this since they wouldn't want to overpay and they will want a fast swap at the same time so there would be no attack vector and if the taker payment transaction takes too much time to confirm due to a fee spike there is always the option of immediate refund. We should allow maker paying less for fees if possible since it will benefit either the maker on success or taker on refund, so taker will validate the fee to be either `taker_instructed_fee` or less.

2. The other solution is more complicated and it's a borrow from lightning where the maker and taker renegotiate the fee if there is a fee spike but this complicates things a lot.

Let's use option one for now and wait until this is tested in a real environment. For fee spikes we can allow the taker to use a bit higher fee from his side (10% higher or we can set it more accurately once we see real data) if needed in the future.